src/types/util.rs

```diff
@@ -15,6 +15,38 @@ pub struct Resolution {
 pub struct Transition {
     /// Duration of a transition in milliseconds.
     pub duration_ms: f64,
+    pub easing_function: Option<EasingFunctionDefinition>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(untagged)]
+pub enum EasingFunctionDefinition {
+    String(EasingFunction),
+    Object { function_name: EasingFunction },
+    ObjectWithParams(EasingFunctionWithParams),
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
+#[serde(tag = "function_name", rename_all = "snake_case")]
+pub enum EasingFunctionWithParams {
+    CubicBezier { points: [f64; 4] },
+}
```

Comment:
I would simplify this:
```suggestion
#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
#[serde(untagged)]
pub enum EasingFunctionDefinition {
    BuiltIn(EasingFunction),
    Custom(CustomCubicBezierEasing),
}

#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema)]
pub struct CustomCubicBezierEasing {
    cubic_bezier_points: [f64; 4],
}
```
such that in API it looks like:
```typescript
type Transition = {
  duration_ms: f64;
  easing_function?: BuiltInEasingFunction | CustomCubicBezierEasing;
}

type BuiltInEasingFunction =
  | "linear"
  | "ease"
  | "ease_in"
  | "ease_in_out"
  | "ease_in_quint"
  | "ease_out_quint"
  | "ease_in_out_quint"
  | "ease_in_expo"
  | "ease_out_expo"
  | "ease_in_out_expo"
  | "bounce"

type CustomCubicBezierEasing = {
    cubic_bezier_points: [f64, f64, f64, f64]
}
```

Reply:
I'd prefer to keep the current API:
```json
{
  "function_name": "linear"
}
```
```json
{
  "function_name": "cubic_bezier",
  "control_points": [0, 0, 1, 1]
}
```

I think this is more consistent.

I will simplify the enums though

Reply:
Ok